src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -775,18 +777,28 @@ public String setInheritSteward(
       UUID datasetId, boolean inheritSteward, AuthenticatedUserRequest userReq) {
     String description =
         String.format("Set inherit steward to %s for dataset %s", inheritSteward, datasetId);
-    var custodianPolicy =
+    List<SamPolicyModel> datasetPolicies =
         iamService.retrievePolicies(userReq, IamResourceType.DATASET, datasetId).stream()
-            .filter(p -> p.getName().equals(IamRole.CUSTODIAN.toString()))
-            .findFirst()
-            .orElseThrow();
+            .filter(
+                p ->
+                    List.of(IamRole.CUSTODIAN.toString(), IamRole.STEWARD.toString())
+                        .contains(p.getName()))
+            .toList();
+    List<String> datasetPolicyEmails =
+        new ArrayList<>(datasetPolicies.stream().map(SamPolicyModel::getEmail).distinct().toList());
+    List<String> datasetPolicyMembers =
+        new ArrayList<>(
+            datasetPolicies.stream()
```

Comment:
I found that I had to initialize these lists as ArrayList in order to support deserialization when pulling these values out of the flight map.